### ISSUE TITLE
Windows 10/Node 10 can't run migrations

### ISSUE THREAD

**Issue by frankdugan3** [graphile/migrate#14]
Getting this error on Windows 10/Node v10.15.1/graphile-migrate v0.0.3 when running migrations:
```
(node:20544) UnhandledPromiseRejectionWarning: Error: Invalid migration header in '${fullPath}'
    at Promise.all.files.filter.map (D:\Programming\hsm\node_modules\graphile-migrate\dist\migration.js:75:19)
(node:20544) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:20544) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with
a non-zero exit code.
```
Tried using PowerShell, CMD, and Git bash, didn't seem to make a difference. Migrations work fine on multiple Linux boxes.

**Comment by frankdugan3** [comment]
A guess: in `migration.js` on line #59: 
```
const fullPath = `${committedMigrationsFolder}/${filename}`;
```
Windows uses `\` as the directory separator, which the `path` lib would abstract away. I'm wondering if maybe string building directly is causing a mismatch between Linux/Windows?

**Comment by ngld** [comment]
This looks more like an issue with linebreak characters. That error message appears if the migration header isn't followed by an empty line. The [code checks](https://github.com/graphile/migrate/blob/master/src/migration.ts#L123) if the second `\n` in the file is followed by another `\n`. This works fine on Unix. However, if you convert the linebreaks to Windows-style, you'll get `\r\n\r\n` which causes the check to fail.
Try manually converting the migration files on Windows with `dos2unix` or disabling git's `core.autocrlf` feature.

**Comment by frankdugan3** [comment]
@ngld Good idea, I’ll report back when I get a chance to try.

**Comment by benjie** [comment]
It's unlikely it's the directory separator, despite all the hoo-ha about it, Windows supports `/` just fine and has done for years and years. I think linebreaks is most likely the culprit as @ngld says; a PR to make us more flexible in our use of linebreaks would be welcome. I actually have Windows now, so at some point I can try this out for myself!

**Comment by benjie** [comment]
If people are using the line ending re-writing features of git (which they may be doing unintentionally) then the hashes will not line up. But I don’t think we can safely replace \r\n with \n because it could be inside a string and hence significant. We need a syntax-aware hash algorithm really. For now; let’s just recommend that Windows users use the above gitattributes; if you felt like sending a PR adding a “Windows” section to the README with the relevant instructions that would be very helpful. 🙏

**Comment by benjie** [comment]
Closing this as I believe it was fixed in #22

**Comment by benjie** [comment]
Out in 0.0.10
